File: src/leetcode/solved-collection/19. Remove Nth Node From End of List/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func removeNthFromEnd(head *ListNode, n int) *ListNode {
    count := 0
    temp := head
    for temp != nil {
        temp = temp.Next
        count++
    }

    index := count - n

    node := &ListNode{}
    node.Next = head
    tmp := node
    for i := 0; i < count; i++ {
        if i == index {
            if tmp.Next != nil {
                tmp.Next = tmp.Next.Next
            } else {
                tmp.Next = nil
            }
        } else if tmp.Next != nil {
            tmp = tmp.Next
        }
    }
    return node.Next
}

File: src/leetcode/solved-collection/19. Remove Nth Node From End of List/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        count = 0
        temp = head
        while temp:
            temp = temp.next
            count += 1

        index = count - n
        node = ListNode()
        node.next = head
        tmp = node
        for i in range(count):
            if i == index:
                tmp.next = tmp.next.next if tmp.next else None
            else:
                tmp = tmp.next

        return node.next
